apollo-compiler/src/main/kotlin/com/apollostack/compiler/ir/Fragment.kt

```diff
@@ -0,0 +1,46 @@
+package com.apollostack.compiler.ir
+
+import com.squareup.javapoet.ClassName
+import com.squareup.javapoet.MethodSpec
+import com.squareup.javapoet.TypeSpec
+import javax.lang.model.element.Modifier
+
+data class Fragment(
+    val fragmentName: String,
+    val source: String,
+    val typeCondition: String,
+    val fields: List<Field>,
+    val fragmentsSpread: List<String>,
+    val inlineFragments: List<String>,
+    val fragmentsReferenced: List<String>) : CodeGenerator {
+  /** Returns a Java method that returns the interface represented by this Fragment object. */
+  private fun toMethodSpec() =
+      MethodSpec.methodBuilder(fragmentName.decapitalize())
+          .returns(ClassName.get("", fragmentName.capitalize()))
+          .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
+          .build()
+
+  /** Returns the Java interface that represents this Fragment object. */
+  override fun toTypeSpec(): TypeSpec =
+      TypeSpec.interfaceBuilder(fragmentName.capitalize())
+          .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+          .addMethods(fields.map(Field::toMethodSpec))
+          .build()
+
+  companion object {
```

Comment:
sorry  I meant about this `genericAccessorMethodSpec`. The intention of this is to not make decision about the name of method that returns fragments here but move to the guy who is actually uses it, in our case `OperationTypeSpecBuilder`